### DISCUSSION TITLE
Expected leadership/periodic jobs behavior with blue/green deploys

### DISCUSSION THREAD

**Discussion by tony-pizza** [sidekiq/sidekiq#6416] [Q&A]
We recently investigated a case of missed executions of periodic jobs that were scheduled to run at the same time. We believe these jobs were missed due to not having a Sidekiq leader at the time the jobs were scheduled, as a result of the leader election process during an unfortunately-timed deploy. We want to make sure our understanding is correct.

Is it true that any time a leader steps down there will be some period of time where Sidekiq is leaderless until a new leader is elected, and during this time periodic jobs will not be scheduled (and won't be "backfilled" to be run by the next leader)?

We've observed that this leaderless window tends to be around 10 seconds, but I'm guessing that this could theoretically be as long as the follower leader verification window (60 seconds).

Thank you!

**Comment by mperham** [comment]
Correct, that’s the reason why the periodic jobs feature explicitly does not implement backfill and why we strongly urge you to build cron jobs which are resilient if a window is missed.
